### PR TITLE
Bugfix FXIOS-7616 [v121] Fix errant log files remaining in Cache

### PR DESCRIPTION
**Pull request by mattreaganmozilla**
## :scroll: Tickets
[Jira ticket](https://mozilla-hub.atlassian.net/browse/FXIOS-7616)
[Github issue](https://github.com/mozilla-mobile/firefox-ios/issues/16956)

## :bulb: Description

This is part of a broader investigation into [FXIOS-7616](https://mozilla-hub.atlassian.net/browse/FXIOS-7616). A few aspects are still being looked into, notably:
- The cause of the app size increase for some users (current evidence points to log files though [browsing history might also be a problem](https://github.com/mozilla-mobile/firefox-ios/issues/16235))
- Why clearing app data does not resolve the problem

The runaway log files were addressed in https://github.com/mozilla-mobile/firefox-ios/pull/16530 though I'm investigating if there are other potential culprits involved. However there is also the problem that users attempting to clear data via `Settings > Data Management` cannot free up this excess space. Only after deleting + reinstalling is this resolved. iOS is notoriously stubborn for not clearing `Cache` folders within the app bundles even in cases where disk space has become limited.

This PR attempts to fix part of the above issue by ensuring that when users clear their Cache data we also clear all log data (which, depending on what bugs may have previously impacted those users, could be very large and possibly not removed correctly by our logging system).

Although [SwiftyBeaver](https://github.com/SwiftyBeaver/SwiftyBeaver) should now be properly rotating log files and limiting their size, it's conceivable that other issues could have caused older log files to remain in `Library/Caches/Logs`. I tested the same steps as Johan from https://github.com/mozilla-mobile/firefox-ios/issues/16956 and verified that data in `/Library/Caches/Logs` is not cleared; this PR ensures that we remove old log data as part of the Cache deletion also.

afterClearing

## :pencil: Checklist
You have to check all boxes before merging
- [x] Filled in the above information (tickets numbers and description of your work)
- [x] Updated the PR name to follow our [PR naming guidelines](https://github.com/mozilla-mobile/firefox-ios/wiki/Pull-Request-Naming-Guide)
- [ ] Wrote unit tests and/or ensured the tests suite is passing
- [ ] When working on UI, I checked and implemented accessibility (minimum Dynamic Text and VoiceOver)
- [ ] If needed I updated documentation / comments for complex code and public methods

### CODE REVIEW

**Comment by mattreaganmozilla** [comment]
@OrlaM I'm still investigating but I suspect this might be part of the problem with https://github.com/mozilla-mobile/firefox-ios/issues/16956 . LMK if you might want to sync up further on any of this in the meantime.

File: BrowserKit/Sources/Common/Logger/LoggerFileManager.swift [hunk 1]
```diff
@@ -37,15 +40,25 @@ class DefaultLoggerFileManager: LoggerFileManager {
...
     func copyLogsToDocuments() {
-        deleteOldLogs()
+        deleteOldDocumentDirectoryLogs()
         copyLogs()
     }
 
     // MARK: - Private
 
+    /// Deletes all log files in Caches (and Documents) directory.
+    func deleteCachedLogFiles() {
```
Comment (lmarceau) [inline BrowserKit/Sources/Common/Logger/LoggerFileManager.swift:50]: Can we add a test case in `LoggerFileManagerTests` for this `deleteCachedLogFiles`?

**Review by lmarceau — COMMENTED** [review]
Nice find!

File: BrowserKit/Sources/Common/Logger/LoggerFileManager.swift (hunk 1 above)
Comment (mattreaganmozilla) [reply BrowserKit/Sources/Common/Logger/LoggerFileManager.swift:50]: Yes good call-out thanks, I added an initial test for now and am taking a closer look at possibly building those out a bit further as well

**Review by thatswinnie — APPROVED** [review]

**Comment by mattreaganmozilla** [comment]
@Mergifyio backport release/v120

**Comment by mattreaganmozilla** [comment]
@Mergifyio backport release/v119

**Pull request merged**